src/client/activeobjectmgr.cpp

```diff
@@ -29,9 +29,8 @@ void ActiveObjectMgr::clear()
 {
 	// delete active objects
 	for (auto &active_object : m_active_objects) {
-		delete active_object.second;
-		// Object must be marked as gone when children try to detach
-		active_object.second = nullptr;
+		// Note: Object must be marked as gone when children try to detach
+		active_object.second.reset();
 	}
 	m_active_objects.clear();
```

Comment:
I was worried because of the comment. But as `aomgr::getActiveObject()` also returns null if the obj is no longer in the container, this should be fine. Will do.

Reply:
Oops, you’re right. Reentrancy is scary. Better restore the loop.
Or use something like this:
```suggestion
	// Empty m_active_objects and *then* destroy the objects so their destructors
	// will see an empty map, rather than a map being emptied
	decltype(m_active_objects) objects_to_destroy;
	std::swap(objects_to_destroy, m_active_objects);
```

Reply:
I don't think reentrancy of this function matters here. Why would an obj destructor clear the objmgr?

However, reentrance of the container (aka reentrence between the containers member functions) is be an issue. It's probably undefined what `m_active_objects.find()` does while `m_active_objects.clear()` or `m_active_objects.erase()` is executed.
I would've overlooked this, thanks!

Clearing all at once however, is more likely to be wrong. Currently all objects that are alive (and are not the one that is currently being destructed) are in the objmgr. On clear, we kill and remove one obj after another. If we instead take out all at once from the objmgr, and then kill the objs, that's different.

I'll add back the loop with an appropriate comment.